contrib/dracut/90zfs/zfs-load-key.sh.in

```diff
@@ -17,10 +17,8 @@
 [ "${root##zfs:}" = "${root}" ] && [ "${root##ZFS=}" = "${root}" ] && [ "$rootfstype" != "zfs" ] && exit 0
 
 # There is a race between the zpool import and the pre-mount hooks, so we wait for a pool to be imported
-while true; do
-    zpool list -H | grep -q -v '^$' && break
-    [ "$(systemctl is-failed zfs-import-cache.service)" = 'failed' ] && exit 1
-    [ "$(systemctl is-failed zfs-import-scan.service)" = 'failed' ] && exit 1
+while [ "$(zpool list -H)" = "" ]; do
+    systemctl is-failed --quiet zfs-import-cache.service zfs-import-scan.service && exit 1
```

Comment:
This check is now a little different since we're relying on the return code instead of the "failed" text.  Is it possible we might now incorrectly exit because `systemctl is-failed` returns non-zero for some other error or warning.

Reply:
It could fail if PID1 has somehow died, which will panic anyway, I guess?
But I'd expect it to return some failure code instead of 0 in that, or any other, failure case that can be encountered in an initrd, so that's not a problem I forsee.